apps/explorer/lib/explorer/chain/token/instance.ex

```diff
@@ -870,4 +870,66 @@ defmodule Explorer.Chain.Token.Instance do
   def media_type_to_string({type, subtype}) do
     "#{type}/#{subtype}"
   end
+
+  @doc """
+  Preloads NFTs for a list of `TokenTransfer` structs.
+
+  ## Parameters
+
+    - `token_transfers`: A list of `TokenTransfer` structs.
+    - `opts`: A keyword list of options.
+
+  ## Returns
+
+  A list of `TokenTransfer` structs with preloaded NFTs.
+  """
+  @spec preload_nft([TokenTransfer.t()] | Transaction.t(), keyword()) :: [TokenTransfer.t()] | Transaction.t()
+  def preload_nft(token_transfers, options) when is_list(token_transfers) do
+    token_instances_id =
+      token_transfers
+      |> Enum.reduce(MapSet.new(), fn
+        %TokenTransfer{token_type: nft_token_type} = token_transfer, ids
+        when nft_token_type in ["ERC-721", "ERC-1155", "ERC-404"] ->
+          MapSet.put(ids, {List.first(token_transfer.token_ids), token_transfer.token_contract_address_hash.bytes})
+
+        _token_transfer, ids ->
+          ids
+      end)
+      |> MapSet.to_list()
+
+    token_instances =
+      Instance
+      |> where(
+        [nft],
+        fragment(
+          "(?, ?) = ANY(?::token_instance_id[])",
+          nft.token_id,
+          nft.token_contract_address_hash,
+          ^token_instances_id
+        )
+      )
+      |> Chain.select_repo(options).all()
+      |> Enum.reduce(%{}, fn nft, map ->
+        Map.put(map, {nft.token_id, nft.token_contract_address_hash}, nft)
+      end)
+
+    Enum.map(token_transfers, fn
+      %TokenTransfer{token_type: "ERC-20"} = token_transfer ->
+        token_transfer
+
+      token_transfer ->
+        %TokenTransfer{
+          token_transfer
+          | token_instance:
+              token_instances[{List.first(token_transfer.token_ids), token_transfer.token_contract_address_hash}]
+        }
+    end)
```

Comment:
Wouldn't it be safer to utilize the same guard as above?

```suggestion
    Enum.map(token_transfers, fn
      %TokenTransfer{token_type: nft_token_type} when nft_token_type in ["ERC-721", "ERC-1155", "ERC-404"]->
        %TokenTransfer{
          token_transfer
          | token_instance:
              token_instances[{List.first(token_transfer.token_ids), token_transfer.token_contract_address_hash}]
        }
      token_transfer ->
        token_transfer
    end)
```